### PR TITLE
Simplify headless- and Raspbian gcc-ARM-build

### PR DESCRIPTION
**Pull request by bagong**
- this moves the two GUI folders in common and JITlib to scide_scqt where it will be ignored in qt-less mode
- adds the compiler flags for armv6l/armv7l as documented in the Raspi/BBB build instructions
- adds brief info to Linux readme

This basically allows to build with `cmake ..` on Raspberry Pi. It doesn't touch the deprecated-3.7 file as there are plans to remove/move that anyways.

### CODE REVIEW

**Comment by timblechmann** [comment]
this breaks using gui from the readline interface or when running sclang scripts from command line. ide and gui are orthogonal.

**Comment by bagong** [comment]
Ah, damn, I didn't think of that! Haha, actually I experienced the problem and thought is was some other misconfiguration on my system. I'll remove that, sorry!
What about the identification of the system(s) to add the flags. Is that correct and worth it?

File: CMakeLists.txt [hunk 1]
```diff
@@ -297,6 +297,25 @@ if(MINGW)
...
 endif()
 
+# support building on Raspberry Pi
+if(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_SYSTEM_PROCESSOR STREQUAL "armv7l")
+  message(STATUS "Adding compiler flags for armv7l (RPi 2&3, BBB)")
+  foreach(flag CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+    set(${flag} "${${flag}} -mtune=cortex-a7")
+    set(${flag} "${${flag}} -mfloat-abi=hard")
+    set(${flag} "${${flag}} -mfpu=neon")
+    set(${flag} "${${flag}} -funsafe-math-optimizations")
```
Comment (timblechmann) [inline CMakeLists.txt]: please double-check that this won't break demand ugens which use NaNs as trigger

Comment (danstowell) [inline CMakeLists.txt]: yeah I'd be dubious about unsafe-math-optimizations. Do you have a specific reason to want it, or are you just being optimisitc? I'd say remove it if no good reason.

File: CMakeLists.txt [hunk 2]
```diff
@@ -297,6 +297,25 @@ if(MINGW)
...
+  endforeach()
+elseif(CMAKE_COMPILER_IS_GNUCXX AND CMAKE_SYSTEM_PROCESSOR STREQUAL "armv6l")
+  message(STATUS "Adding compiler flags for armv6l (RPi 1b)")
```
Comment (danstowell) [inline CMakeLists.txt]: These status messages are not needed. Fine for your own development but not to merge...

File: CMakeLists.txt (hunk 1 above)
Comment (bagong) [inline CMakeLists.txt]: Just following @redFrik, he probably wanted to tweak the very slow RP1. Thanks for your input, both of you, will correct!

**Comment by redFrik** [comment]
i just tried to compile and compare sc master both with and without the flag (-funsafe-math-optimizations).  did benchmarks and various tests on a rpi3 running jessie - also with various demand ugens - but couldn't tell any difference.  i have now forgotten where i found the recommendation to set this flag.  thanks for noting.

**Comment by bagong** [comment]
Done. So if this is welcome and you'd prefer a cleaner history, I can redo the PR in it's latest state.

**Comment by danstowell** [comment]
@bagong thanks; yes I'd say redo the PR, because we don't want to risk those `git mv` confusing the history in future...

**Comment by crucialfelix** [comment]
ControlModel.sc should not be included. Back in 2000 it was originally designed for faders, but the Spec and ControlSpec classes are used for all types of value mapping. Not just gui.

**Comment by crucialfelix** [comment]
is Model included ? there are legitimate subclasses of that which don't need or want to know if a gui exists or not. 

so this will mean we can compile without getting a 70MB binary, right ? and easily moveable. kewl!

**Comment by bagong** [comment]
The class-library is unchanged (I reverted the move), people still depend on moving the GUI folders manually. @redFrik  (I think) already removed model.sc from the gui folders on master in the past, so moving or deleting the GUI folders is a bit simpler. 

The high-tech solution proposed in #1209 is still required, sorry, I thought too short, the only thing that changes is an a bit easier compile and a few hints in the Readme.

On Windows I made a workaround in moving the GUI folders as a post install step if SC_QT is off. But that may be considered clumsy and seems to require a rather recent cmake version.

**Comment by redFrik** [comment]
yes, the Model.sc file moved to SCClassLibrary/Common/Control/ in sc3.8
https://github.com/supercollider/supercollider/pull/1976/commits/4d835b79b8c8d437423913c4fd565ae181cf7185  this was because some other core classes (Buffer) needed it.
perhaps ControlModel.sc should move as well, but i haven't run into any code that complains about it missing so far (on headless systems).
moving (or deleting if you want to save space!) gui classes manually isn't such a big thing so i'm fine with leaving them there.  also the manual move is only needed on jessie-lite, beaglebone console and similar system that completely lacks a windows manager.

**Comment by bagong** [comment]
Should I move ControlModel to Common/Control as well as part of this?

**Comment by bagong** [comment]
Thanks for the input!!

**Comment by crucialfelix** [comment]
Even if a jedi like @redFrik think's it's easy, for other people it's mysterious and liable to make them just give up. So this is a very useful enhancement. thanks for doing it!

**Pull request closed without merging**